### PR TITLE
CLDR-15653 improve LocaleCompletion

### PR DESCRIPTION
**Pull request by srl295**
- fix typos
- add comments
- remove dead code

CLDR-15653

- [ ] This PR completes the ticket.

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletion.java [hunk 1]
```diff
@@ -128,9 +128,8 @@ static final class LocaleCompletionHelper implements Listener {
         LocaleCompletionHelper() {
             phf = PathHeader.getFactory(CLDRConfig.getInstance().getEnglish());
             cache =  CacheBuilder.newBuilder().maximumSize(500)
-            // We would want the expiry time greater if this were used by users, not just TC.
-            .expireAfterWrite(15, TimeUnit.MINUTES)
-            // .concurrencyLevel()
+            .concurrencyLevel(5) // allow 5 threads to compute completion, uncontested
+            .expireAfterWrite(Duration.ofMinutes(20)) // expire 20 min after last change
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletion.java:132]: interesting!

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletion.java [hunk 2]
```diff
@@ -166,8 +163,10 @@ static LocaleCompletionResponse handleGetLocaleCompletion(final CLDRLocale cldrL
         logger.info("Starting LocaleCompletion for " + options.toString());
         final TestResultBundle checkCldr = stFactory.getTestResult(cldrLocale, options);
 
-        // we need an XML Source to receive notification
-        final XMLSource mySource = stFactory.makeSource(localeId);
+        // we need an XML Source to receive notification.
+        // This causes LocaleCompletionHelper.INSRTANCE.valueChanged(...) to be called
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletion.java:167]: typo: INSRTANCE

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletion.java [hunk 3]
```diff
@@ -166,8 +163,10 @@ static LocaleCompletionResponse handleGetLocaleCompletion(final CLDRLocale cldrL
...
+        // whenever a vote happens.
+        final XMLSource mySource = stFactory.makeSource(localeId, false);
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletion.java:169]: false means "not resolved"?

**Review by btangmu — APPROVED** [review]

**Review by macchiati — DISMISSED** [review]

**Comment by btangmu** [comment]
@srl295 there's a new bug involving locale completion (3rd meter): https://unicode-org.atlassian.net/browse/CLDR-15662

it's likely to need changes to LocaleCompletion.java -- could we merge this PR now to avoid conflicts? is the failing build something else? I'll try to restart the build

**Comment by btangmu** [comment]
the failed test was unrelated -- TestUserLevel

**Comment by srl295** [comment]
> the failed test was unrelated -- TestUserLevel

yes it was the one i fixed earlier.  

Rebasing - please rereview (no change) and we can merge once it's passing

**Comment by srl295** [comment]
@btangmu  merge if you want if i miss it

**Comment by srl295** [comment]
@btangmu  also note https://unicode-org.atlassian.net/browse/CLDR-15357 (i linked it)

**Review by btangmu — APPROVED** [review]

**Pull request merged**